src/qwircl/core.cljs

```diff
@@ -43,13 +54,90 @@
 ;;       ; Draw the circle.
 ;;       (q/ellipse x y 100 100))))
 
+(defn set-color [color]
+  (apply q/fill 
+         (condp = color 
+           :green [0 120 0]
+           :blue [135 206 250]
+           :purple [120 0 120]
+           :red [120 0 0]
+           :orange [255 165 0]
+           :yellow [255 255 0]
+           :black [0 0 0]
+           ; [0 255 255]
+           background-color)))
+
+(defn draw-black-background [x y]
+  (set-color :black)
+  (q/rect (* size x) (* size y) size size))
+
+(defn draw-tile [x y tile]
+  (let [middle (/ size 2)
+        quarter (* size 0.25)
+        eighth (* 0.5 quarter)]
+    (draw-black-background x y)
+    (set-color (tile :color))
+    (q/no-stroke)
+    (q/with-translation [(* size x) (* size y)]
+      (condp = (tile :shape)
+        :clover (do
+                  (q/ellipse middle (/ size 3) quarter quarter)
+                  (q/ellipse (/ size 3) middle quarter quarter)
+                  (q/ellipse middle (* size (/ 2 3)) quarter quarter)
+                  (q/ellipse (* size (/ 2 3)) middle quarter quarter)
+                  (q/ellipse middle middle quarter quarter))
+        :star (do 
+                (q/triangle middle 0 
+                            size (* 0.75 size)
+                            0 (* 0.75 size))
+                (q/triangle middle size
+                            0 (* 0.25 size)
+                            size (* 0.25 size)))
+        :cross (do
+                 (q/quad quarter 0
+                         (- size quarter) 0
+                         (- size quarter) size
+                         quarter size)
+                 (q/quad 0 quarter
+                         0 (- size quarter)
+                         size (- size quarter)
+                         size quarter))
+        :circle (q/ellipse middle middle (* 0.75 size) (* 0.75 size))
+        :diamond (q/quad middle quarter 
+                         (- size quarter) middle
+                         middle (- size quarter)
+                         quarter middle)
+        :rectangle (q/rect (/ middle 2) (/ middle 2) middle middle)))
+    ; (q/stroke 0 0 0)
+    (q/no-stroke)
+    (set-color :background)))
+
+(defn draw-empty-space [x y]
+  (set-color :background)
+  (q/rect (* size x) (* size y) size size))
+
+(defn draw-grid [grid]
+  (q/with-translation [0 header]
+    (doseq [x (range tiles) 
+            y (range tiles)]
+      (if-let [tile (get-in grid [x y])]
+        (draw-tile x y tile)
+        (draw-empty-space x y)))))
+
+(defn draw-header [player]
+  (q/fill 0 0 0)
+  (q/text (:name player) 10 header)
+  (dotimes [x (count (:hand player))]
+    (draw-tile x 0 ((:hand player) x)))
+  (apply q/fill background-color))
+
 (defn draw-state [state]
+  (q/frame-rate 120)
   (q/background 240)
-  (q/fill 0 255 255)
-  (q/stroke 0 0 0)
-  (doseq [x (range tiles) 
-          y (range tiles)]
-    (q/rect (* size x) (* size y) size size)))
+  (apply q/fill background-color)
```

Comment:
`q/fill` has a unary overload that takes an rgb vector. I think `set- color` uses `apply` too

Reply:
Done.